packages/gatsby/src/redux/reducers/inference-metadata.js

```diff
@@ -17,8 +17,8 @@ const ignoredFields = new Set([
 
 module.exports = (state = {}, action) => {
   switch (action.type) {
-    case `DELETE_CACHE`:
-      return {}
+    // case `DELETE_CACHE`:
```

Comment:
Weird bug or accident in deleting?

Reply:
Currently a bug because inference metadata (recently added for schema rebuilding) is affected by clearing cache 

@vladar How do we fix this?

Reply:
There is a bunch of options actually:

1. We could send a list of nodes to be deleted along with DELETE_CACHE action and handle it in the metadata reducer similar to how we handle DELETE_NODE
2. Or maybe run DELETE_NODE for all dirty nodes instead of DELETE_CACHE
3. Defer metadata generation to the end of the bootstrap (maybe don't even save it in a cache in this case). This only works if this DELETE_CACHE action cannot occur outside of the bootstrap.

In general, I think it is a good idea to dispatch nodes to be purged via redux dispatch loop so that any other reducers could act as they need.